### PR TITLE
[PHOENIX-3623] Integrate Omid with Phoenix.

### PR DESCRIPTION
**Pull request by ohadshacham**
This patch add's support for Omid transaction processor.
More information exists at the Jira and at [OMID-82], which is the corresponding jira and commit to Omid.

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/transaction/OmidTransactionProvider.java [hunk 1]
```diff
@@ -104,10 +204,15 @@ public Provider getProvider() {
     @Override
     public boolean isUnsupported(Feature feature) {
         return true;
+        // return (
+        //   feature == Feature.ALTER_NONTX_TO_TX || 
+        //   feature == Feature.COLUMN_ENCODING || 
+        //   feature == Feature.MAINTAIN_LOCAL_INDEX_ON_SERVER || 
+        //   feature == Feature.SET_TTL);
```
Comment (JamesRTaylor) [inline phoenix-core/src/main/java/org/apache/phoenix/transaction/OmidTransactionProvider.java]: Please remove commented out code.

File: phoenix-core/src/main/java/org/apache/phoenix/transaction/OmidTransactionContext.java [hunk 2]
```diff
@@ -18,89 +18,279 @@
...
+import org.apache.phoenix.exception.SQLExceptionCode;
+import org.apache.phoenix.exception.SQLExceptionInfo;
 import org.apache.phoenix.jdbc.PhoenixConnection;
 import org.apache.phoenix.schema.PTable;
 import org.apache.phoenix.transaction.TransactionFactory.Provider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+//import org.apache.omid.tso.TSOMockModule;
```
Comment (JamesRTaylor) [inline phoenix-core/src/main/java/org/apache/phoenix/transaction/OmidTransactionContext.java]: Remove commented out import

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/OmidTransactionalProcessor.java [hunk 3]
```diff
@@ -17,13 +17,16 @@
...
+import org.apache.omid.transaction.OmidSnapshotFilter;
+import org.apache.phoenix.transaction.OmidTransactionProvider;
 
 
 public class OmidTransactionalProcessor extends DelegateRegionObserver {
 
     public OmidTransactionalProcessor() {
-        super(new BaseRegionObserver());
+        // Hack for testing - retrieves the commit table client from the singleton OmidTransactionProvider
+        // TODO: use real commit table and get port from config
```
Comment (twdsilva) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/OmidTransactionalProcessor.java]: Does this comment need to be addressed?

File: phoenix-core/src/main/java/org/apache/phoenix/transaction/OmidTransactionProvider.java [hunk 4]
```diff
@@ -63,26 +83,106 @@ public PhoenixTransactionContext getTransactionContext(PhoenixConnection connect
...
+
+        Injector injector = Guice.createInjector(new TSOMockModule(tsoConfig));
+        tso = injector.getInstance(TSOServer.class);
+        tso.startAndWait();
+
+        OmidClientConfiguration clientConfig = new OmidClientConfiguration();
+        clientConfig.setConnectionString("localhost:" + port);
+        clientConfig.setConflictAnalysisLevel(OmidClientConfiguration.ConflictDetectionLevel.ROW);
+
+        InMemoryCommitTable commitTable = (InMemoryCommitTable) injector.getInstance(CommitTable.class);
```
Comment (twdsilva) [inline phoenix-core/src/main/java/org/apache/phoenix/transaction/OmidTransactionProvider.java]: Do we always use an InMemoryCommitTable? or is this just for tests?

**Comment by MohammedLayeeq** [comment]
Hi Team,
Integrate Omid with Phoenix 
how we can  integrate  both of this service.
is this integration supports Phoenix 5.0.0  or not.

**Comment by stoty** [comment]
Already merged.

**Pull request closed without merging**